### ISSUE TITLE
Check failed: host_index <= host_buffer->size() (20138 vs. 10069) in cudnn_conv_algorithm_picker.cc:284

### ISSUE THREAD

**Issue by byronyi** [tensorflow/tensorflow#29350]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian 9.8
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 9628bc838b9f8f4a5abc2f9f76808f2c212c3182
- Python version: 3.5
- Bazel version (if compiling from source): 0.25.2
- GCC/Compiler version (if compiling from source): 6.3
- CUDA/cuDNN version: 10.0/7.4.1
- GPU model and memory: V100-SXM2-16GB

**Describe the expected behavior**

The benchmark script finishes normally and prints performance numbers, as only a single-day of commits before on 2019-05-31 with 2d47a0ebca181a9476cbd6d971c819d05fd6f7e1 (we run into this with our in-house untested nightly build at 2019-06-01).

**Describe the current behavior**

```
I0603 16:27:54.804590 140049462265600 session_manager.py:500] Running local_init_op.
I0603 16:28:02.836288 140049462265600 session_manager.py:502] Done running local_init_op.
2019-06-03 16:28:11.377907: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
2019-06-03 16:28:14.911938: F tensorflow/compiler/xla/service/gpu/cudnn_conv_algorithm_picker.cc:284] Check failed: host_index <= host_buffer->size() (20138 vs. 10069)
Fatal Python error: Aborted

Thread 0x00007f5fce730700 (most recent call first):
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1429 in _call_tf_sessionrun
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1341 in _run_fn
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1356 in _do_call
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1350 in _do_run
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1173 in _run
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 950 in run
  File "/benchmarks/scripts/tf_cnn_benchmarks/benchmark_cnn.py", line 868 in benchmark_one_step
  File "/benchmarks/scripts/tf_cnn_benchmarks/benchmark_cnn.py", line 2429 in benchmark_with_session
  File "/benchmarks/scripts/tf_cnn_benchmarks/benchmark_cnn.py", line 2293 in _benchmark_graph
  File "/benchmarks/scripts/tf_cnn_benchmarks/benchmark_cnn.py", line 2084 in _benchmark_train
  File "/benchmarks/scripts/tf_cnn_benchmarks/benchmark_cnn.py", line 1879 in run
  File "benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py", line 68 in main
  File "/usr/local/lib/python3.5/dist-packages/absl/app.py", line 251 in _run_main
  File "/usr/local/lib/python3.5/dist-packages/absl/app.py", line 300 in run
  File "benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py", line 72 in <module>
```

**Code to reproduce the issue**

Running https://github.com/tensorflow/benchmarks with the following options:

```
python3 benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py \
    --num_gpus=8 \
    --model=resnet50 \
    --batch_size=256 \
    --use_fp16 \
    --variable_update=replicated \
    --all_reduce_spec=nccl \
    --xla
```

or (the error message is all the same)

```
python3 benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py \
    --num_gpus=8 \
    --model=resnet50 \
    --batch_size=256 \
    --use_fp16 \
    --variable_update=parameter_server \
    --local_parameter_device=cpu \
    --xla
```

**Other info / logs**

Our Bazel build command is attached below:

```
bazel build \
    -c opt \
    --action_env CUDA_TOOLKIT_PATH="/usr/local/cuda" \
    --action_env GCC_HOST_COMPILER_PATH="/usr/bin/gcc" \
    --action_env=PYTHON_BIN_PATH="/usr/bin/python3" \
    --action_env=PYTHON_LIB_PATH="/usr/local/lib/python3.5/dist-packages" \
    --action_env=TF_CONFIGURE_IOS="0" \
    --action_env=TF_CUDA_COMPUTE_CAPABILITIES="6.1,7.0,7.5" \
    --config=cuda \
    --config=gdr \
    --config=tensorrt \
    --config=verbs \
    --copt=-Wno-sign-compare \
    --copt=-march=ivybridge \
    --define=with_default_optimizations=true \
    --define=with_xla_support=true \
    --host_copt=-march=ivybridge \
    --python_path="/usr/bin/python3" \
    //tensorflow/tools/pip_package:build_pip_package
```

**Comment by byronyi** [comment]
Friendly ping @tfboyd

**Comment by byronyi** [comment]
Ping @timshen91 for this suspicious CL: e734bb03dd8a188b040c527f3fcfce9f6ec85631

**Comment by byronyi** [comment]
Using a single GPU will not reveal this error.

**Comment by byronyi** [comment]
Removing the `static` specifier in https://github.com/tensorflow/tensorflow/commit/e734bb03dd8a188b040c527f3fcfce9f6ec85631#diff-a471413a17d71b1d18f51d6cf23e0029R278 resolves the issue.

I need to sign another corporate CLA though, and I don't mind anyone else open a PR with this fix while I am waiting for the process to be done.

**Comment by tfboyd** [comment]
I just opened an internal bug. I will link this to it and hopefully get it merged today.

**Comment by tfboyd** [comment]
Internal bug that is also linked to this one:  https://b.corp.google.com/issues/134386554

**Comment by byronyi** [comment]
Should be obsoleted by 3679396fb5676fed06d84a35c63fe29bd257a829.

**Comment by byronyi** [comment]
@tfboyd Reopen as this issue still affects r2.0. Ping @bananabowl @martinwicke for cherry-picking 3679396, and ping @timshen91 to double check.

**Comment by bananabowl** [comment]
+ Goldie for r2.0

**Comment by byronyi** [comment]
Should be fixed by 93eee337a1932744df77515dc855dc478b4609a4. Thanks @goldiegadde!
